redis/src/cluster_async/mod.rs

```diff
@@ -1811,48 +1826,46 @@ where
                         "The user connection for node {:?} is unhealthy. Error: {:?}",
                         node, err
                     );
-                    Some(conn_type)
+                    Some(RefreshConnectionType::OnlyUserConnection)
                 }
             }
         }
-        RefreshConnectionType::OnlyManagementConnection => {
-            let mut management_conn = node.management_connection.clone().await;
-            match check_connection(&mut management_conn, timeout).await {
-                Ok(_) => None,
-                Err(err) => {
+        RefreshConnectionType::OnlyManagementConnection => match management_conn_state.await {
+            Ok(_) => None,
+            Err(err) => {
+                warn!(
```

Comment:
maybe move the warning to 
```rs
    let management_conn_state = async {
        match node.management_connection.clone() {
            Some(conn) => check_connection(&mut conn.await, timeout).await,
            None => Err(RedisError::from((
                ErrorKind::NoneManagementConn,
                "The management connection isn't set",
            ))),
        }
    };
    ```
    instead of repeating it twice.